kratos/utilities/auxilary_global_master_slave_relation.h

```diff
@@ -0,0 +1,213 @@
+ *   MasterSlaveConstraint then there will only be one entry for master for its corresponding AuxilaryGlobalMasterSlaveRelation.
+ *   Currently this class is designed to hold only one equation. There is only one unique object of this class for each slave. 
+ *
+ *   Future plan is to also make it possible to work with matrices (T) and vectors (for slave and master equation ids and constants)
+ *
+ *
+ *  IMPORTANT : This is not seen by the user. This is a helper data structure which is exists only in the builder and solver. 
+ *
+ * @author Aditya Ghantasala
+ */
+class AuxilaryGlobalMasterSlaveRelation : public IndexedObject
+{
+  public:
+    typedef std::size_t IndexType;
+    typedef Matrix MatrixType;
+    typedef Vector VectorType;
+    typedef std::vector<IndexType> EquationIdVectorType;
+    KRATOS_CLASS_POINTER_DEFINITION(AuxilaryGlobalMasterSlaveRelation);
+
+    // empty constructor and methods to add master and slave independently.
+    AuxilaryGlobalMasterSlaveRelation() : IndexedObject(0)
+    {
+        mConstant = 0.0;
+    }
+
+    AuxilaryGlobalMasterSlaveRelation(IndexType const &rSlaveEquationId) : IndexedObject(rSlaveEquationId)
+    {
+        mConstant = 0.0;
+    }
+
+    /**
+     * Function to get the slave equation Id corresponding to this constraint.
+     * @param Constant the value of the constant to be assigned.
+     */
+    IndexType SlaveEquationId() const { return this->Id(); }
+
+    /**
+     * Function to set the lefthand side of the constraint (the slave dof value)
+     * @param LhsValue the value of the lhs (the slave dof value)
+     */
+    void SetLHSValue(double const &LhsValue)
+    {
+        mLhsValue = LhsValue;
+    }
+
+    /**
+     * Function to update the righthand side of the constraint (the combination of all the master dof values and constants)
+     * @param RhsValue the value of the lhs (the slave dof value)
+     */
+    void SetRHSValue(double const &RhsValue) { mRhsValue = RhsValue; }
+    void UpdateRHSValue(double const &RhsValueUpdate)
+    {
+        mRhsValue += RhsValueUpdate;
+    }
+
+    // Get number of masters for this slave
+    IndexType GetNumberOfMasters() const
+    {
+        return mMasterEquationIdVector.size();
+    }
+
+    /**
+     * this determines the master equation IDs connected to this constraint
+     * @param rResult the elemental equation ID vector
+     * @param rCurrentProcessInfo the current process info instance
+     */
+    virtual void EquationIdVector(IndexType &rSlaveEquationId,
+                                  EquationIdVectorType &rMasterEquationIds,
+                                  ProcessInfo &rCurrentProcessInfo)
+    {
+        if (rMasterEquationIds.size() != 0 || rMasterEquationIds.size() == 0)
+            rMasterEquationIds.resize(this->GetNumberOfMasters(), false);
+
+        rSlaveEquationId = this->SlaveEquationId();
+
+        for (IndexType i = 0; i < this->GetNumberOfMasters(); i++)
```

Comment:
++i

Reply:
done ... but it does not matter here though ...